### PR TITLE
platform checks: Enable RBAC checks

### PR DESCRIPTION
**Pull request by def-**
As suggested in https://github.com/MaterializeInc/materialize/pull/18477#discussion_r1152885371

Part of https://github.com/MaterializeInc/materialize/issues/17983

### Motivation

  * This PR adds a known-desirable feature.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: misc/python/materialize/checks/mzcompose_actions.py [hunk 1]
```diff
@@ -68,6 +69,29 @@ def execute(self, e: Executor) -> None:
...
+
+                $ postgres-execute connection=mz_system
+                ALTER SYSTEM SET max_tables = 1000;
+                ALTER SYSTEM SET max_sinks = 1000;
+                ALTER SYSTEM SET max_sources = 1000;
+                ALTER SYSTEM SET max_materialized_views = 1000;
+                ALTER SYSTEM SET max_objects_per_schema = 1000;
+                ALTER SYSTEM SET max_secrets = 1000;
+
+                # We already want to test with RBAC enabled, but give materialize user
```
Comment (philip-stoev) [inline misc/python/materialize/checks/mzcompose_actions.py:87]: I think this comment belongs above, at `ALTER ROLE materialize...` ?

File: misc/python/materialize/checks/scenarios.py [hunk 2]
```diff
@@ -64,6 +65,12 @@ def base_version(self) -> MzVersion:
         return self._base_version
 
     def run(self) -> None:
+        actions = self.actions()
+        assert isinstance(
+            actions[0], StartMz
```
Comment (philip-stoev) [inline misc/python/materialize/checks/scenarios.py:70]: Is this true in cloudtest as well? It does not use `StartMz`

File: misc/python/materialize/checks/scenarios.py [hunk 3]
```diff
@@ -64,6 +65,12 @@ def base_version(self) -> MzVersion:
...
+        ), "Expecting the actions to start by starting Materialize"
```
Comment (philip-stoev) [inline misc/python/materialize/checks/scenarios.py:71]: "The list of actions must start with StartMz"?

File: misc/python/materialize/checks/drop_table.py [hunk 4]
```diff
@@ -35,6 +35,12 @@ def manipulate(self) -> List[Testdrive]:
                 > DROP TABLE drop_table1;
                 """,
                 """
+                # When upgrading from old version without roles the talbe is
```
Comment (philip-stoev) [inline misc/python/materialize/checks/drop_table.py:38]: table

**Comment by def-** [comment]
Have to test them together to see if everything is green now, so put these two changes into same PR.

**Review by philip-stoev — APPROVED** [review]

**Review by jkosh44 — APPROVED** [review]
LGTM

File: misc/python/materialize/checks/rename_view.py [hunk 5]
```diff
@@ -27,6 +28,19 @@ def initialize(self) -> Testdrive:
...
+                # When upgrading from old version without roles the views are
+                # owned by default_role, thus we have to change the owner
+                # before dropping them:
+                $ postgres-execute connection=postgres://mz_system:materialize@materialized:6877
+                ALTER VIEW rename_view_viewB2 OWNER TO materialize;
+                ALTER VIEW rename_view_viewA2 OWNER TO materialize;
+                """
+            if self.base_version >= MzVersion.parse("0.46.0-dev")
+            else ""
+        )
```
Comment (nrainer-materialize) [inline misc/python/materialize/checks/rename_view.py:42]: Optional: Just wondering if it makes sense to extract common parts to a generic function. 🤔 What do you think?
If this makes sense, could we either automatically identify the tables and views to be adjusted or hand over these objects to a generic `fix_ownership` function / method?

**Comment by def-** [comment]
@philip-stoev I'm a bit confused with how to make sure the new `ConfigureMz` action runs in the beginning in parallel mode

File: misc/python/materialize/checks/rename_view.py (hunk 5 above)
Comment (nrainer-materialize) [reply misc/python/materialize/checks/rename_view.py:42]: Alright 👍

**Comment by philip-stoev** [comment]
> @philip-stoev I'm a bit confused with how to make sure the new `ConfigureMz` action runs in the beginning in parallel mode.

You need to make sure that the testdrive that is being used to run your ALTERs is being waited on to completion, something like:

```
diff --git a/misc/python/materialize/checks/scenarios.py b/misc/python/materialize/checks/scenarios.py
index c9bfe4af3..085a579b3 100644
--- a/misc/python/materialize/checks/scenarios.py
+++ b/misc/python/materialize/checks/scenarios.py
@@ -69,7 +69,7 @@ class Scenario:
         # The first action is StartMz, configure it implicitly afterwards
         actions.insert(1 if isinstance(actions[0], StartMz) else 0, ConfigureMz(self))
         for action in actions:
-            action.execute(self.executor)
+            action.execute(self.executor).join()
 ```
 
You may also need to `def join` for the StartMz action to be a noop, otherwise it is going to assert.
This way, if `Executor` e is parallel, that executor's testdrive() implementation will run testdrive in a thread:

**Pull request merged**